src/btree/bt_cursor.c

```diff
@@ -350,17 +361,19 @@ __wt_cursor_valid(WT_CURSOR_BTREE *cbt, WT_UPDATE **updp, bool *valid)
             return (0);
 
         /* Check for an update. */
-        if (page->modify != NULL && page->modify->mod_row_update != NULL) {
-            WT_RET(__wt_txn_read(session, cbt, page->modify->mod_row_update[cbt->slot], &upd));
-            if (upd != NULL) {
-                if (upd->type == WT_UPDATE_TOMBSTONE) {
-                    if (F_ISSET(upd, WT_UPDATE_RESTORED_FROM_DISK))
-                        __wt_free_update_list(session, &upd);
-                    return (0);
-                }
-                if (updp != NULL)
-                    *updp = upd;
+        WT_RET(__wt_txn_read(session, cbt,
```

Comment:
Even a page with no modifications might need to lookaside search to find an update. In that case, just pass `NULL` and we'll go straight to checking on-disk and lookaside.